eng/Generate.ps1

```diff
@@ -134,7 +134,7 @@ function Add-Directory ([string]$testName, [string]$directory, [boolean]$forTest
     $readmeConfigurationPath = Join-Path $directory "readme.md"
     $testArguments = $null
     if (Test-Path $readmeConfigurationPath) {
-        $testArguments = "--require=$readmeConfigurationPath --clear-output-folder=true"
+        $testArguments = "--require=$readmeConfigurationPath --clear-output-folder=true --generate-test-scaffolding=true"
```

Comment:
I add this flag for a few reasons:
1. Generation of the test cases is slightly different from generation of samples. For instance, the test cases need to instrument the client and client options, therefore we have to use a ctor with a client options, this is different from samples (in which we always use the simplest ctor). Therefore generation of the test cases have the possibility that it cannot compile. This flag helps our customer to mitigate the issue if present.
2. Our customer should never modify the generated test cases because they are generated. The build analyze pipeline on azure-sdk-for-net repo validates if they get modified. Therefore these code snippets are only references for them to write their own test cases. They could copy them and then change them. If they copy them and finish their own test based on the code we generate, they no longer need the generated test cases any more.

Here is the work flow I imagine for this feature:
1. They generate the code with this flag on, and get the test cases. (the flag is default off)
2. They copy the test cases and modify them to finalize the test cases.
3. They generate the code with this flag off, the generated test cases are gone.

Reply:
Yes, customer should never modify the generated code, and we put the generated test code under test\Generated. And customer will add more test under `test` directory. Current they can copy them to `test` directory and modify them. The final goal is to auto-gen test which is completed and workable by default. Current we can set this flag false by default because the quality is not good enough. In future, we will change it to `true` by default ( or maybe remove this flag). 
 How about change it to `--generate-test` without `scaffolding`?

Reply:
OK I could rename them

Reply:
This flag is renamed to `--generate-tests` to align with one of our existing build target in mgmt (`/t:GenerateTests`)